client_code/MultiSelectDropDown/__init__.py

```diff
@@ -169,6 +171,17 @@ def selected(self, values):
     def change(self, *e):
         return self.raise_event("change")
 
+    def shown(self, *e):
+        self._menu_shown = True
+
+    def hidden(self, *e):
+        self._menu_shown = False
```

Comment:
I don't think we want shown and hidden to be public method here.

Consider creating them in the init method
```python
    def __init__(self, **properties):
        ...
        self._menu_visible = False

        def shown(*e):
            self._menu_visible = True
        self._el.on("shown.bs.select", shown)

        def hidden(*e):
            self._menu_visible = False
        self._el.on("hidden.bs.select", hidden)

```